src/actions/bookingDaysOffActions.js

```diff
@@ -60,119 +43,158 @@ const validateBody = (body) => {
+});
+};
 
-const validateUserExists = (body, users) => {
+const checkUserExistsInDB = userId => {
     return new Promise((resolve, reject) => {
-        if (findItemById(users, body.userId) === undefined) {
+        let findUser = User.findById(userId, (err, user) => {
+            if (err){
+                reject({
+                    status: 400,
+                    message: "Bad user id"
+                });
+                return 
+            }
+            if (user) resolve();
             reject({
-                "status": 404,
-                "message": "User not found"
+                status: 404,
+                message: "User not found"
             });
-		}
-        resolve();
-    })
-}
+        });
+    });
+};
 
 const formatDate = date => {
     if (date < 10) return `0${date}`;
     else return String(date);
-}
+};
 
 const daysOffRangeToArray = (startDate, endDate) => {
     return new Promise((resolve, reject) => {
         startDate = new Date(startDate);
         endDate = new Date(endDate);
-
+        
         let daysOffArray = [];
-
+        
         if (startDate > endDate) {
             reject({
-                "status": 400,
-                "message": "End date lower than start date"
+                status: 400,
+                message: "End date lower than start date"
             });
         }
-
+        
         endDate.setDate(endDate.getDate() + 1);
         for (; startDate < endDate; startDate.setDate(startDate.getDate() + 1)) {
             if (startDate.getDay() > 0 && startDate.getDay() < 6) {
-                daysOffArray.push(`${startDate.getFullYear()}-${formatDate(startDate.getUTCMonth() + 1)}-${formatDate(startDate.getDate())}`);
+                daysOffArray.push(
+                    `${startDate.getFullYear()}-${formatDate(
+                        startDate.getUTCMonth() + 1
+                    )}-${formatDate(startDate.getDate())}`
+                );
             }
         }
-
+        
         resolve(daysOffArray);
-    })
+    });
+};
+
+const createUpdatedArray = (alreadyBookedDays, arrayOfDaysOff) => {
+    let updatedDaysOff = [];
+    for(let i=0; i<arrayOfDaysOff.length; i++){
+        if(alreadyBookedDays.indexOf(arrayOfDaysOff[i]) === -1) 
+            updatedDaysOff.push(arrayOfDaysOff[i])
+    }
+    return updatedDaysOff
 }
 
-const createNewDaysOffJSON = (daysOff, userId, daysOffArray) => {
-    return new Promise((resolve, reject) => {
-        resolve({
-            id: newId(daysOff),
-            userId: userId,
-            daysOff: daysOffArray
-        });
+const removeDuplicateDaysOff = (userId,arrayOfDaysOff) => {
+   return new Promise((resolve,reject) => {
+        getAlreadyBookedDays(userId)
+            .then(alreadyBookedDays => {
+                updatedDaysOff = createUpdatedArray(alreadyBookedDays, arrayOfDaysOff)
+                if(updatedDaysOff.length === 0){
+                    reject({
+                        status: 422,
```

Comment:
We generally prefer 40x status codes, but it's ok to use any if there is a strong reason for it

Reply:
nu ni s-a parut ca se incadreaza intr-un status cod 40x si 422 am crezut noi ca e adecvat. Care ar fi mai potrivit aici?

Reply:
It's ok to use any status code you want, as long as you have in your mind a clear reason for doing so. If you do, than just summarise it here and consider this comment resolved 😄

Reply:
Puteti folosi orice status code atat timp cat va e clar motivul pentru care il folositi si il explicati si aici in PR